engine/src/test/java/dgca/verifier/app/engine/DefaultAffectedFieldsDataRetrieverTest.kt

```diff
@@ -0,0 +1,534 @@
+            "        \"co\": \"AT\",\n" +
+            "        \"ci\": \"URN:UVCI:V1:DE:JYP9MNRWCMLK8B20328XRYUEV:\",\n" +
+            "        \"mp\": \"EU/1/20/1507\",\n" +
+            "        \"is\": \"asfadf\",\n" +
+            "        \"sd\": 2,\n" +
+            "        \"tg\": \"840539006\"\n" +
+            "      }\n" +
+            "    ],\n" +
+            "    \"nam\": {\n" +
+            "      \"fnt\": \"STANDARDISEDFAMILY\"\n" +
+            "    },\n" +
+            "    \"ver\": \"1.3.0\",\n" +
+            "    \"dob\": \"\"\n" +
+            "  }\n" +
+            "}"
+
+    private val RULE = "{\n" +
+            "  \"Identifier\": \"VR-DE-0001\",\n" +
+            "  \"Type\": \"Acceptance\",\n" +
+            "  \"Country\": \"DE\",\n" +
+            "  \"Version\": \"1.0.0\",\n" +
+            "  \"SchemaVersion\": \"1.0.0\",\n" +
+            "  \"Engine\": \"CERTLOGIC\",\n" +
+            "  \"EngineVersion\": \"0.7.5\",\n" +
+            "  \"CertificateType\": \"Vaccination\",\n" +
+            "  \"Description\": [\n" +
+            "    {\n" +
+            "      \"lang\": \"en\",\n" +
+            "      \"desc\": \"The vaccination schedule must be complete (e.g., 1/1, 2/2).\"\n" +
+            "    },\n" +
+            "    {\n" +
+            "      \"lang\": \"de\",\n" +
+            "      \"desc\": \"Die Impfreihe muss vollständig sein (z.B. 1/1, 2/2).\"\n" +
+            "    },\n" +
+            "    {\n" +
+            "      \"lang\": \"fr\",\n" +
+            "      \"desc\": \"La série vaccinale doit être complète (p. ex. 1/1, 2/2).\"\n" +
+            "    },\n" +
+            "    {\n" +
+            "      \"lang\": \"es\",\n" +
+            "      \"desc\": \"La pauta de vacunación debe estar completa (por ejemplo, 1/1, 2/2).\"\n" +
+            "    },\n" +
+            "    {\n" +
+            "      \"lang\": \"it\",\n" +
+            "      \"desc\": \"Il ciclo di vaccinazione deve essere stato completato (ad es. 1/1, 2/2).\"\n" +
+            "    }\n" +
+            "  ],\n" +
+            "  \"ValidFrom\": \"2021-07-03T00:00:00Z\",\n" +
+            "  \"ValidTo\": \"2030-06-01T00:00:00Z\",\n" +
+            "  \"AffectedFields\": [\n" +
+            "    \"v.0\",\n" +
+            "    \"v.0.dn\",\n" +
+            "    \"v.0.sd\"\n" +
+            "  ],\n" +
+            "  \"Logic\": {\n" +
+            "    \"if\": [\n" +
+            "      {\n" +
+            "        \"var\": \"payload.v.0\"\n" +
+            "      },\n" +
+            "      {\n" +
+            "        \">=\": [\n" +
+            "          {\n" +
+            "            \"var\": \"payload.v.0.dn\"\n" +
+            "          },\n" +
+            "          {\n" +
+            "            \"var\": \"payload.v.0.sd\"\n" +
+            "          }\n" +
+            "        ]\n" +
+            "      },\n" +
+            "      true\n" +
+            "    ]\n" +
+            "  }\n" +
+            "}"
```

Comment:
Could we use `trimIdent()` here to improve readability and make it easier to adapt the json if it changes?
```kotlin
    private val RULE = """
        {
            "Identifier": "VR-DE-0001",
            "Type": "Acceptance",
            "Country": "DE",
            "Version": "1.0.0",
            "SchemaVersion": "1.0.0",
            "Engine": "CERTLOGIC",
            "EngineVersion": "0.7.5",
            "CertificateType": "Vaccination",
            "Description": [
                {
                    "lang": "en",
                    "desc": "The vaccination schedule must be complete (e.g., 1/1, 2/2)."
                },
                {
                    "lang": "de",
                    "desc": "Die Impfreihe muss vollständig sein (z.B. 1/1, 2/2)."
                },
                {
                    "lang": "fr",
                    "desc": "La série vaccinale doit être complète (p. ex. 1/1, 2/2)."
                },
                {
                    "lang": "es",
                    "desc": "La pauta de vacunación debe estar completa (por ejemplo, 1/1, 2/2)."
                },
                {
                    "lang": "it",
                    "desc": "Il ciclo di vaccinazione deve essere stato completato (ad es. 1/1, 2/2)."
                }
            ],
            "ValidFrom": "2021-07-03T00:00:00Z",
            "ValidTo": "2030-06-01T00:00:00Z",
            "AffectedFields": [
                "v.0",
                "v.0.dn",
                "v.0.sd"
            ],
            "Logic": {
                "if": [
                    {
                        "var": "payload.v.0"
                    },
                    {
                        ">=": [
                            {
                                "var": "payload.v.0.dn"
                            },
                            {
                                "var": "payload.v.0.sd"
                            }
                        ]
                    },
                    true
                ]
            }
        }
    """.trimIndent()
```